bots/running-a-bbot-bot/README.md

```diff
@@ -0,0 +1,184 @@
+prototyping with a free cloud hosting environment but aren't suitable for a bot
+that will serve heavy traffic on an ongoing basis.
+
+## Deploy Docker Container
+
+The included `Dockerfile` is all that's required for configuring the build.
+
+If your environment requires first building the Docker image (e.g. local test),
+you can do that as follows, though many cloud environments will detect the
+Docker setup and do this automatically. Many services will allow for defining
+environment variables in their own dash UI too, so the `-e` lines below might
+not be required either.
+
+> NOTE: The Docker build copies `./index.js` and `./src` to the built image. No
+> other scripts are included, so keep all your bot customisation within `src`.
+
+Build a docker image:
+
+```
+docker build -t bbot:rocketchat .
+```
+
+Run the docker image, passing environment configs:
+
+```
+docker run -it --rm --name brocket \
+  -e ROCKETCHAT_URL=<YOUR_URL> \
+  -e ROCKETCHAT_USER=<BOT_USER> \
+  -e ROCKETCHAT_PASSWORD=<BOT_PASS> \
+  -e ROCKETCHAT_ROOM='' \
+  -e LISTEN_ON_ALL_PUBLIC=true \
+  -e RESPOND_TO_EDITED=true \
+  -e RESPOND_TO_DM=true \
+  -e RESPOND_TO_LIVECHAT=true \
+  bbot:rocketchat
+```
+
+## Deploy to Heroku
+
+Heroku can detect the Node app type and automatically run the main script.
+
+You should install the [Heroku CLI][heroku-cli] and login. Either create your
+Heroku app following the linked instructions or add an existing app to your
+bot's git as a remote host, per instructions below.
+
+It just needs the environment configuration:
+
+```
+heroku config:set ROCKETCHAT_URL=<YOUR_URL>
+heroku config:set ROCKETCHAT_USER=<BOT_USER>
+heroku config:set ROCKETCHAT_PASSWORD=<BOT_PASS>
+heroku config:set ROCKETCHAT_ROOM=''
+heroku config:set LISTEN_ON_ALL_PUBLIC=true
+heroku config:set RESPOND_TO_DM=true
+heroku config:set RESPOND_TO_EDITED=true
+heroku config:set RESPOND_TO_LIVECHAT=true
+```
+
+Then add your Heroku app as a git remote and push to release:
+
+```
+heroku git:remote -a <YOUR_APP>
+git push heroku master
+```
+
+## Deploy Docker to Heroku
+
+Log in to Container Registry
+
+```
+heroku container:login
+```
+
+Build the image and push to Container Registry:
+
+```
+heroku container:push web
+```
+
+Then release the image to your app:
+
+```
+heroku container:release web
+```
+
+Now open the app in your browser:
+
+```
+heroku open
+```
+
+## Deploy to Glitch
+
+Remix the [Glitch boilerplate][brocket-glitch].
+
+Edit **.env** as per example:
+
+```
+ROCKETCHAT_URL=<YOUR_URL>
+ROCKETCHAT_USER=<BOT_USER>
+ROCKETCHAT_PASSWORD=<BOT_PASS>
+RESPOND_TO_DM=true
+RESPOND_TO_EDITED=true
+LISTEN_ON_ALL_PUBLIC=false
+```
+
+That's it, the app will restart and attempt connection when you change the
+settings, just view the logs to resolve any connection issues.
+
+Glitch apps will usually sleep after 5 minutes, but
+[there are ways](support.glitch.com/t/not-letting-the-app-go-down/2000/13)
```

Comment:
This link needs the `http://` prefix, as it is now it breaks the link and the tests.

Reply:
Will fix now, thanks @MartinSchoeler